standard/PortalStatistics.lua

```diff
@@ -0,0 +1,1134 @@
+			type = 'bar',
+			data = StatisticsPortal._addArrays(yearSeriesData),
+		}
+
+		config.yAxis = {type = 'value', name = 'Earnings ($USD)'}
+		config.xAxis = {type = 'category', name = config.catLabel, data = opponentNames}
+		config.customLegend = config.customLegend or config.customInputs
+		return StatisticsPortal._drawChart(config, chartData)
+	end
+end
+
+
+--[[
+Section: Coverage Breakdown
+]] --
+
+
+function StatisticsPortal.coverageStatistics(args)
+	args = args or {}
+	args.alignSide = Logic.readBoolOrNil(args.alignSide)
+
+	local wrapper = mw.html.create('div')
+	local TournamentTable = wrapper:tag('div')
+	local matchTable = wrapper:tag('div')
+
+	if args.alignSide then
+		TournamentTable
+			:addClass('template-box')
+			:css('padding-right','2em')
+		matchTable
+			:addClass('template-box')
+			:css('padding-right','2em')
+	end
+
+	TournamentTable:node(StatisticsPortal.coverageTournamentTable(args))
+	matchTable:node(StatisticsPortal.coverageMatchTable(args))
+
+	return wrapper
+end
+
+
+function StatisticsPortal.coverageMatchTable(args)
+	args = args or {}
+	args.multiGame = Logic.readBoolOrNil(args.multiGame) or false
+	args.customGames = (type(args.customGames) == 'table' and args.customGames)
+		or (String.isNotEmpty(args.customGames) and Array.map(mw.text.split(args.customGames, ',', true), String.trim))
+		or GAMES
+
+	local matchTable = mw.html.create('table')
+		:addClass('wikitable wikitable-striped')
+
+	matchTable:tag('caption')
+		:wikitext(args.matchTableTitle or (Logic.readBool(args.alignSide) and '<br>' or ''))
+		:css('text-align','center')
+
+	local matchHeader = matchTable:tag('tr')
+
+	if Logic.readBool(args.multiGame) then
+		matchHeader:tag('th')
+			:wikitext('Game')
+	end
+
+	matchHeader
+		:tag('th'):wikitext(args.matchesTitle or 'Matches'):done()
+		:tag('th'):wikitext(args.gamesTitle or 'Games')
+
+	if Logic.readBool(args.multiGame) then
+		for _, game in Table.iter.spairs(args.customGames) do
+			matchTable:node(StatisticsPortal._coverageMatchTableRow(args, {game = game, year = args.year}))
+		end
+	end
+
+	matchTable:node(StatisticsPortal._coverageMatchTableRow(args, {year = args.year}))
+
+	return matchTable
+end
+
+
+function StatisticsPortal._coverageMatchTableRow(args, parameters)
+	local resultsRow = mw.html.create('tr')
+	local tagType = (Logic.readBool(args.multiGame) and not parameters.game) and 'th' or 'td'
+
+	if Logic.readBool(args.multiGame) then
+		resultsRow:node(StatisticsPortal._returnGameCell(args, parameters, tagType))
+	end
+
+	resultsRow:tag(tagType)
+		:wikitext(LANG:formatNum(Count.matches(parameters)))
+		:css('text-align','right')
+
+	resultsRow:tag(tagType)
+		:wikitext(LANG:formatNum(Count.games(parameters)))
+		:css('text-align','right')
+
+	return resultsRow:allDone()
+end
+
+
+function StatisticsPortal.coverageTournamentTable(args)
+	args = args or {}
+	args.multiGame = Logic.readBoolOrNil(args.multiGame) or false
+	args.customGames = String.isNotEmpty(args.customGames) and
```

Comment:
you are basically using this plus the next line in at least 4 places
imo make it a helper function that can just be called
```lua
function StatisticsPortal._splitOrDefault(input, default)
	if String.isEmpty(input) then
		return default
	end

	return Array.map(mw.text.split(input, ',', true), String.trim)
end
```
you then can just do this:
```suggestion
	args.customGames = StatisticsPortal._splitOrDefault(args.customGames, GAMES)
```